src/app/Http/Controllers/CrudController.php

```diff
@@ -198,11 +198,30 @@ public function show($id)
     {
         $this->crud->hasAccessOrFail('show');
 
+        // set columns from db
+        $this->crud->setFromDb();
+
+        // cycle through old columns for removal
+        foreach($this->crud->columns as $old_id => $column) {
+            // replace any relationship columns
+            if(array_key_exists('model', $column)) {
+                $this->crud->columns[$column['name']] = $this->crud->columns[$old_id];
+            }
+            // remove old numbered columns
+            if(is_integer($old_id)) {
```

Comment:
What if a column IS called "1" i.e. the number 1?

Reply:
Fixed, thanks!